CourseApp.Tests/DemoTest.cs

```diff
@@ -5,10 +5,48 @@ namespace CourseApp.Tests
 {
     public class DemoTest
     {
+        [Theory]
+        [InlineData(0, 0, double.NaN)]
+        [InlineData(1, 1, double.PositiveInfinity)]
+        [InlineData(1, 2, 1.965)]
+        [InlineData(1, 3, 2.097)]
+        [InlineData(0, -6, 1.727)]
+        [InlineData(3, -3, 6562.097)]
+        [InlineData(5, -3, 390626.097)]
+        [InlineData(46, -2, 97336.965)]
+
+        public void TestCalcAllZeros(double a,  double x, double exp)
+        {
+            var actualResult = Program.Calc(a, x);
+            Assert.Equal(exp, actualResult, 3);
+        }
+
+        [Fact]
+        public void TestTaskA()
+        {
+            var res = Program.TaskA(2, 1, 3, 1);
```

Comment:
Please add additional changes with xn > xk, dx > xk - xn, and empty array for case B